src/index.ts

```diff
@@ -14,16 +14,27 @@
  * limitations under the License.
  */
 
-/* eslint require-await: 0 */
-'use strict'
+import {
+  default as customPluginLib,
+  CustomService,
+  DecoratedFastify,
+  ServiceConfig,
+} from '@mia-platform/custom-plugin-lib'
 
-import {DecoratedFastify} from "@mia-platform/custom-plugin-lib";
-const customService = require('@mia-platform/custom-plugin-lib')()
+import { getMetrics } from './config/metrics'
+import { EnvironmentVariables, environmentSchema } from './config/env'
+import { BaseService } from './interfaces/service'
 
-module.exports = customService(async function index(service: DecoratedFastify) {
+const customService: CustomService<EnvironmentVariables> = customPluginLib(environmentSchema)
+
+module.exports = customService(async(service: DecoratedFastify<ServiceConfig>) => {
```

Comment:
is `module.exports` still needed? can't we use `export` keyword?

Reply:
Yes, `module.exports` is still needed. Otherwise `lc39` is not able to find the exported function.

This is the error:

    Error: The module doesn’t export an async function

I suppose it is an issue with `lc39`. However, I would leave such change to another PR.